src/main/java/seedu/system/model/person/CustomDate.java

```diff
@@ -38,39 +38,16 @@ public static boolean isValidDate(String date) {
         return false;
     }
 
-    /**
-     * Returns true if dateTwo is after the current CustomDate object.
-     *
-     */
-
-    public boolean before(CustomDate dateTwo) {
-        String[] dateOneArr = this.toString().split("/");
-        String[] dateTwoArr = dateTwo.toString().split("/");
-
-        int dayOne = Integer.valueOf(dateOneArr[0]);
-        int dayTwo = Integer.valueOf(dateTwoArr[0]);
-
-        int monthOne = Integer.valueOf(dateOneArr[1]);
-        int monthTwo = Integer.valueOf(dateTwoArr[1]);
-
-        int yearOne = Integer.valueOf(dateOneArr[2]);
-        int yearTwo = Integer.valueOf(dateTwoArr[2]);
-
-        if (yearOne < yearTwo) {
-            return true;
-        } else if (yearOne == yearTwo) {
-            if (monthOne > monthTwo) {
-                return false;
-            }
-            if (dayOne > dayTwo) {
-                return false;
-            }
-            return true;
+    public Date getDate() {
+        try {
+            return new SimpleDateFormat(DATE_FORMAT).parse(date);
+        } catch (ParseException e) {
+            return null;
         }
```

Comment:
getDate() should return dateObj;

the parsing should be done in the constructor

this.dateObj = SimpleDateFormat(DATE_FORMAT).parse(date);

not in getDate()